docs/proposals/shared-quota.md

```diff
@@ -0,0 +1,81 @@
+	// many more less active projects).  These projects will contend on object creation through
+	// this resource.
+	Selector map[string]string
+
+	// Spec defines the desired quota
+	Quota kapi.ResourceQuotaSpec
+}
+
+type ClusterResourceQuotaStatus struct {
+	// Overall defines the actual enforced quota and its current usage across all namespaces
+	Overall kapi.ResourceQuotaStatus
+
+	// ByNamespace slices the usage by namespace.  This division allows for quick resolution of 
+	// deletion reconcilation inside of a single namespace without requiring a recalculation 
+	// across all namespaces.  This map can be used to pull the deltas for a given namespace.
+	ByNamespace map[string]kapi.ResourceQuotaStatus
+}
+```
+
+
+## Enforcement
+`ClusterResourceQuota` will be enforced with an admission plugin that re-uses the bucketing and evalution code from the `ResourceQuota` admission plugin.
+On a per-API server basis, the `ClusterResourceQuota` updates will be locked in-process.  On a per-cluster basis, the `ClusterResourceQuota` updates will be locked by condition updates on `resourceVersion`.
+
+This introduces contention when resources are being created across namespaces.  For example, crq-a covers ns-1 and ns-2.  ns-1 and ns-2 are both creating pods.
+Before the pods are admitted, crq-a is checked.  ns-1 pod creates are blocked waiting for ns-2 pods to complete admission.
+
+
+## APIs
+A new role called `project-owner` will be created.  `project-owners` are like `project-admins`, but they also have the power to update *some* of their `resourcequota` documents.
+We don't want to allow a `project-owner` to modify any `resourcequota` document, because doing that would prevent a `cluster-admin` from being able to "lock down" a namespace by putting very low `resourcequota` into it.
+In the short term, this will enforced by naming a series of "special" quota names that match today's scope: terminating, notTerminating, bestEffort, notBestEffort, default.
+In the longer term, we could support permissions based on label selectors.
+
+Cluster-scoped `resourcequotaallocation` resources which apply to a particular project will be projected into namespaces under the namespace-scoped  `localresourcequota` resource.
+This allows the ACL rule inside of a project to be "get, list" on "localresourcequota".  That way a user can predict whether his resource creation will succeed or fail based on clusterresourcequota.
+
+
+## Comparison to Label Constrained Quota Allocation
+By bounding allocation (quota allocation), we can write a reasonably efficient implementation that only watches resourcequotas across the cluster.  This eliminates cross namespace contention during resource creation.
+Unfortunately, quota allocation can be difficult to use.  Take new project creation as an example.  In order to be safe with respect to the allocation, new projects would have to be created with all quotaed objects set to zero.
```

Comment:
> @deads2k what if someone create project with quota not zero?

Then the allocator will break.  It doesn't change the state of what is, only future changes.  If the new project flow always created a project that allowed one pod (as a for instance), it would allow a user to bypass their allocation limit by creating new projects.

This is distinct from the clusterresourcequota case where such a project would exist, but the user couldn't create anything in it.